src/lib/input/autosize.ts

```diff
@@ -0,0 +1,79 @@
+import {Directive, ElementRef, Input, OnInit} from '@angular/core';
+
+
+/**
+ * Directive to automatically resize a textarea to fit its content.
+ */
+@Directive({
+  selector: 'textarea[md-autosize]',
+  host: {
+    '(input)': 'resizeToFitContent()',
+    '[style.min-height]': '_minHeight',
+    '[style.max-height]': '_maxHeight',
+  },
+})
+export class MdTextareaAutosize implements OnInit {
+  /** Minimum number of rows for this textarea. */
+  @Input() minRows: number;
+
+  /** Maximum number of rows for this textarea. */
+  @Input() maxRows: number;
```

Comment:
could you also do an @Input() for value so you can resize when the content is changed programmatically?